plugin.py

```diff
@@ -66,6 +66,21 @@ def binary_path(cls) -> str:
         else:
             return os.path.join(cls.basedir(), "omnisharp", "OmniSharp.exe")
 
+    @classmethod
+    def get_solution_arguments(cls) -> List[str]:
+        view = sublime.active_window().active_view()
```

Comment:
Where is this `view` variable used? Can it be removed?

Reply:
I see now it's used to get the `window()`, but that can just as well be achieved with `sublime.active_window()`.

Reply:
There is no guarantee that the active window is the one where server is being initialized though.

Reply:
True. This code should go into `on_pre_start`

Reply:
Fair point about the `view` variable! That's what happens when you just copy code from one place to another...

About moving the code to `on_pre_start`, what do you mean exactly? Moving all this there and storing the `solution_file_path` in a member to just read it here?

Reply:
In the `on_pre_start` classmethod you can fill in the final `command` using your logic. You have access to the `window` there as it is passed in the method. This way you don't have to rely on `sublime.active_window()`.

What @rchl mentions about sublime projects not necessarily living alongside the source code is valid. Would it make sense to instead rely on string template substitutions to resolve the .sln file?

As in: I want to be able to specify

```
    "solution_file": "$folder/testconsoleprj.sln"
```

in my .sublime-project. This removes the magic from what relative paths mean as a bonus. You can use template variable substitution in the ususal way with `sublime.expand_variables` and `window.extract_variables()`